tsdb/head.go

```diff
@@ -1202,12 +1202,11 @@ func (h *Head) getOrCreateWithID(id, hash uint64, lset labels.Labels) (*memSerie
 	h.numSeries.Inc()
 
 	h.symMtx.Lock()
-	defer h.symMtx.Unlock()
-
 	for _, l := range lset {
 		h.symbols[l.Name] = struct{}{}
 		h.symbols[l.Value] = struct{}{}
 	}
+	h.symMtx.Unlock()
```

Comment:
I think `h.postings.Add` dependant on this lock in some way. This change (having postings.Add under this lock) was made due to some race that was found. I will verify this again.

Reply:
Looking at the commit that fixed that issue, I can't tell if the fix was moving `h.postings.Add` after the population of the symbols or moving it inside the lock: https://github.com/prometheus/prometheus/commit/823b218e1b2e992338188314c9307437fb075c93

Reply:
@bwplotka you worked on the fix at #7560 I think.
Do you recall whether it was necessary to lock symbols and postings together?
ISTM the bug was that a posting would appear before the symbols were updated.
Can you resurrect your tests that helped find the bug?

Reply:
> ISTM the bug was that a posting would appear before the symbols were updated.

Yes I guess that is the case. In that case we don't need symMtx to cover postings, but postings to come after symbols update.

Reply:
Could you please add a comment about the ordering here since we are taking it outside lock?